code/__HELPERS/path.dm

```diff
@@ -135,34 +137,46 @@
 	src.mintargetdist = mintargetdist
 	src.simulated_only = simulated_only
 	src.avoid = avoid
+	src.skip_first = skip_first
 	src.diagonal_safety = diagonal_safety
+	src.on_finish = on_finish
+
+/datum/pathfind/Destroy(force, ...)
+	. = ..()
+	SSpathfinder.active_pathing -= src
+	SSpathfinder.currentrun -= src
+	if(on_finish)
+		on_finish.Invoke(null)
```

Comment:
Not necessarily. Callbacks don't create any lingering references by default, and so do not define Destroy.
This means in the standard case it's safe to just let them self clean up, like gas mixtures.

This is especially valid in this case, because a callback can be reused, and we have no reason to get in the way of that.
Especially because if it isn't reused it'll just free on its own